### PR TITLE
Added run-length-encoding exercise.

### PR DESCRIPTION
**Pull request by cpaulbond**

### CODE REVIEW

File: exercises/run-length-encoding/run-length-encoding-test.el [hunk 1]
```diff
@@ -0,0 +1,61 @@
+;;; run-length-encoding-test.el --- Tests for run-length-encoding (exercism)
+
+;;; Commentary:
+
```
[the remark this answers is not in the document]
Comment (benreyn) [reply exercises/run-length-encoding/run-length-encoding-test.el]: I picked this up from the ruby track and I think its a good practice. We ill always know for a specific exercise if we need to update the test suite.

File: exercises/run-length-encoding/example.el [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+;;; Commentary:
+
+;;; Code:
+
+
+(defun run-length-group (s)
+  "return list of character run counts"
+  (let ((case-fold-search nil)
+	groups)
+    (with-temp-buffer
```
Comment (benreyn) [inline exercises/run-length-encoding/example.el]: Could this be accomplished with a data structure in memory rather than a temp-buffer?

Comment (cpaulbond) [reply exercises/run-length-encoding/example.el]: Sure, that is always an option.  In fact, my first try did use an in-memory structure but it was more complex.  This version is much simpler, and it pushes the work of grouping letters down into the regular expression code so it should be faster.   Simpler and faster seams like a win to me.

Comment (benreyn) [reply exercises/run-length-encoding/example.el]: Alright, Sounds like a good enough argument to me 👍

**Comment by benreyn** [comment]
Im going to go ahead and merge this as is. I have it on my personal to-do list to add the canonical data versioning to the comments on all the test files, so ill just get this one when I do those others.

**Pull request merged**